virtool/shutdown.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+import aiohttp.web
+
+import virtool.startup
+
+logger = logging.getLogger(__name__)
+
+
+async def exit_client(app: aiohttp.web.Application):
+    """
+    Attempt to close the async HTTP client session.
+
+    :param app: The application object
+    """
+    logger.info("Stopping HTTP client")
+
+    try:
+        await app["client"].close()
+    except KeyError:
+        pass
+
+
+async def exit_dispatcher(app: aiohttp.web.Application):
+    """
+    Attempt to close the app's `Dispatcher` object.
+
+    :param app: The application object
+    """
+    logger.info("Stopping dispatcher")
+
+    try:
+        await app["dispatcher"].close()
+    except KeyError:
+        pass
+
+
+async def exit_executor(app: aiohttp.web.Application):
+    """
+    Attempt to close the app's `ThreadPoolExecutor`.
+
+    :param app: The application object
+    """
+    try:
+        app["executor"].shutdown(wait=True)
+    except KeyError:
+        pass
+
+
+async def exit_process_executor(app: aiohttp.web.Application):
+    """
+    Attempt to close the app's `ProcessPoolExecutor`.
+
+    :param app: The application object
+    """
+    try:
+        app["process_executor"].shutdown(wait=True)
+    except KeyError:
+        pass
```

Comment:
Combine to mirror `init_executors`.
```suggestion
async def exit_executors(app: aiohttp.web.Application):
    """
    Attempt to close the `ThreadPoolExecutor` and `ProcessPoolExecutor`.

    :param app: the application object
    """
    try:
        app["executor"].shutdown(wait=True)
    except KeyError:
        pass
    try:
        app["process_executor"].shutdown(wait=True)
    except KeyError:
        pass
```